### PR TITLE
Week5part4

### PR DESCRIPTION
**Pull request by rivki770**

### CODE REVIEW

File: harry_is_rational.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import os
+import re
+import codecs
+
+
+def rename_files(path):
```
Comment (itayday) [inline harry_is_rational.py:6]: Please add documantation to the function.

File: harry_is_rational.py [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    list_of_file = os.listdir(path)
```
Comment (talbarfm) [inline harry_is_rational.py:7]: llist_of_files*

Comment (talbarfm) [inline harry_is_rational.py:7]: does listdir return only files? cant it return something else as well?

File: harry_is_rational.py [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+    for entry in list_of_file:
```
Comment (talbarfm) [inline harry_is_rational.py:8]: nice variable name :)

File: harry_is_rational.py [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+        full_path = os.path.join(path, entry)
+        codecs.register_error("strict", codecs.ignore_errors)
```
Comment (talbarfm) [inline harry_is_rational.py:10]: While the builtin [open()](https://docs.python.org/3/library/functions.html#open) and the associated [io](https://docs.python.org/3/library/io.html#module-io) module are the recommended approach for working with encoded text files, this module provides additional utility functions and classes that allow the use of a wider range of codecs when working with binary files:
from here - https://docs.python.org/3/library/codecs.html

Comment (talbarfm) [reply harry_is_rational.py:10]: why did you use codecs?

File: harry_is_rational.py [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+        f = codecs.open(full_path, 'r').read()
```
Comment (talbarfm) [inline harry_is_rational.py:11]: use full variable names - not a single character variable names

File: harry_is_rational.py [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+        match = re.search('<title>(.*?)</title>', f)
+        title = match.group(1)
+        title = title.split("Chapter ")  #we get title[o] = before chapter, title[1] = after chapter
```
Comment (talbarfm) [inline harry_is_rational.py:14]: Add spaces after # , add these comments on a separate line above this code

File: harry_is_rational.py [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+        title = title[1].split(":")  #we get title[o] = a number, title[1] = a title
+
+        title[0] = title[0].zfill(3) #conver the number to number with 3 digits
+
+        full_path_new = os.path.join(path, title[0] + title[1] + ".html")
+        os.rename(full_path, full_path_new)
+
+
+list_file = rename_files("C:\\rivki\\studies\\YearD\\SemesterB\\Excellent\\Python\\HW_1\\Git\\yam-mesika-weeks-5-6-rivki770\\potter")
```
Comment (talbarfm) [inline harry_is_rational.py:23]: Add a newline at the end of the file

Comment (talbarfm) [inline harry_is_rational.py:23]: https://www.freecodecamp.org/news/if-name-main-python-example/

File: tools.py [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+    for p in pointer:
+        arr.append(p)
+    print(arr)
+
+    #example_3
+    arr = []
+    pointer = interleave('abc', [1, 2, 3], ('!', '@', '#'), 'XYZ')
+    for p in pointer:
+        arr.append(p)
+    print(arr)
```
Comment (talbarfm) [inline tools.py:34]: So much dup code
So much dup code
So much dup code
So much dup code
maybe export it to a function and call it a few times? :)

File: tools.py [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+    current_number = 0
+    while current_number < size_min:
+        for i in iterable:
+            yield i[current_number]
+        current_number = current_number + 1
+
+
+def main():
+    #example_1
+    arr = []
```
Comment (talbarfm) [inline tools.py:16]: https://www.w3schools.com/python/python_arrays.asp
arrays are not supported in python, lists are :)

File: tools.py [hunk 10]
```diff
@@ -0,0 +1,39 @@
...
+    pointer = interleave('abc', [1, 2, 3], ('!', '@', '#'), 'XYZ')
+    for p in pointer:
+        arr.append(p)
+    print(arr)
+
+
+
+if __name__ == '__main__' :
+    main()
```
Comment (talbarfm) [inline tools.py:39]: Add a newline at eof

**Review by talbarfm — CHANGES_REQUESTED** [review]

**Pull request closed without merging**